internal/cli/atlas/search/create.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/json"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+type CreateOpts struct {
+	cli.GlobalOpts
+	clusterName    string
+	name           string
+	dbName         string
+	collection     string
+	analyzer       string
+	searchAnalyzer string
+	dynamic        bool
+	fields         []string
+	store          store.SearchIndexCreator
+}
+
+func (opts *CreateOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *CreateOpts) Run() error {
+	index, err := opts.newSearchIndex()
+	if err != nil {
+		return err
+	}
+	result, err := opts.store.CreateSearchIndexes(opts.ConfigProjectID(), opts.clusterName, index)
+	if err != nil {
+		return err
+	}
+
+	return json.PrettyPrint(result)
+}
+
+func (opts *CreateOpts) newSearchIndex() (*atlas.SearchIndex, error) {
+	f, err := opts.indexFields()
+	if err != nil {
+		return nil, err
+	}
+	i := &atlas.SearchIndex{
+		Analyzer:       opts.analyzer,
+		CollectionName: opts.collection,
+		Database:       opts.dbName,
+		Mappings: &atlas.IndexMapping{
+			Dynamic: opts.dynamic,
+			Fields:  &f,
+		},
+		Name:           opts.name,
+		SearchAnalyzer: opts.searchAnalyzer,
+	}
+	return i, nil
+}
+func (opts *CreateOpts) indexFields() (map[string]atlas.IndexField, error) {
+	if len(opts.fields) == 0 {
+		return nil, nil
+	}
+	fields := make(map[string]atlas.IndexField, len(opts.fields))
+	for _, p := range opts.fields {
+		f := strings.Split(p, ":")
+		if len(f) != 3 {
+			return nil, fmt.Errorf("partition should be fieldName:analyzer:fieldType, got: %s", p)
+		}
+		fields[f[0]] = atlas.IndexField{
+			Analyzer: f[1],
+			Type:     f[2],
+		}
+	}
+	return fields, nil
+}
+
+// mongocli atlas cluster(s) search|fts index(es) create <indexName> [--clusterName clusterName][--db dbName][--collection collection][--dynamic][--field field][--projectId projectId]
```

Comment:
[np] `[--clusterName clusterName][--db dbName][--collection collection]` -> `--clusterName clusterName --db dbName --collection collection` as they are required?

Reply:
I think this approach in general needs a revamp, to begin with is not appropriate go doc, it should start with the method name I started it as a way to remember what flags I needed as I coded but let me know what you think of what I've done now